src/adapter-types/src/lib.rs

```diff
@@ -73,6 +73,7 @@
 #![warn(clippy::disallowed_macros)]
 #![warn(clippy::disallowed_types)]
 #![warn(clippy::from_over_into)]
+#![warn(clippy::large_futures)]
```

Comment:
Very exited about moving these to the workspace `Cargo.toml` and getting out of the business of having to touch every crate when we adjust a lint :)